### ISSUE TITLE
Incorrect bracket highlighting in markdown code fenced area with JSON syntax

### ISSUE THREAD

**Issue by lewisl** [facelessuser/BracketHighlighter#531]
Please read and fill out this template by replacing the instructions with appropriate information.  If the template is not followed, the issue will be marked `Invalid` and closed.

Before submitting an issue search past issues and read the area of the [documentation](http://facelessuser.github.io/BracketHighlighter/) related to your specific question, issue, or request.

---

## Description

Bracket highlighting error in code fence with JSON syntax

Here it is:
### Example

```json
{
    "layerdefs": {
        {"layer": 2,
            "n_hid": 80,
            "units": "relu"
        },
        {"layer": 1,
            "input": 9999999999999999
        },
        {"layer": 3,
            "classify": "softmax"
        },
    },
    "rundefs": {
        "epochs":  24,
        "alpha":   0.74,
        "reg":  "Maxnorm",
        "maxnorm_lim": [4.0],
        "lambda":  0.000191,
        "learn_decay": [0.5,4.0],
        "mb_size_in":   50, 
        "norm_mode":   "none",
        "dobatch": true,
        "do_batch_norm":  true,
        "opt":   "adam",
        "opt_params": [0.9, 0.999],
        "dropout": false,
        "droplim": [1.0,0.8,1.0],
        "plots": ["Train", "Learning", "Test"],
        "plot_now": true,
        "quiet": true,
        "initializer": "xavier",
        "shuffle": false
    }
}
```

The opening brackets for layer: 1 and layer: 2 are highlighted in red as incorrect.  But, they match properly.  Visual Studio Code gets it right.  Typora gets it right.  Only Sublime Text fails on this.

## Support Info

...

> Run the following command from the menu: `Preferences->Package Settings->BracketHighlighter->Support Info`.  Post the result here.

- ST ver.: 3207
- Platform: osx
- Arch: x64
- Plugin ver.: 2.27.6
- Install via PC: True
- mdpopups ver.: 3.4.0
- backrefs ver.: 3.5
- markdown ver.: 2.6.11
- pygments ver.: 2.1a0
- jinja2 ver.: 2.8

## Steps to Reproduce Issue

1. First step...
Type ``` <return>

2. Second step...
Type some json stuff and end with <return>
then type ``` on a blank line, followed by a blank line

3. Third step...
go back to the ``` at the top and type "json" (without the quotes)

> Provide steps to reproduce the issue. Pictures are fine, but also provide code/text I can copy and paste in order to reproduce. Omit for feature requests and feature proposals.

**Comment by lewisl** [comment]
Intriguingly, github highlighting also gets it wrong.

**Comment by facelessuser** [comment]
1. BracketHighlighter does not syntax highlight (that is the red you are seeing).
2. The JSON data is malformed as the last entry in a hash should not have a trailing comma. That is according to the JSON spec.
3. Please show an example of BracketHighlighter not matching correct brackets as I was able to match all of them with BracketHighlighter despite syntax highlighter showing red.

**Comment by lewisl** [comment]
That doesn’t fix it for me.  I’ll edit my issue and see if github markdown unmarks the braces.  Give me a sec…

**Comment by facelessuser** [comment]
Your JSON is still malformed.

You have a hash `{}` with no keys, you should be using a list with `[]`. It should be an array. Again, this has nothing to do with BracketHighlighter:

```json
{
    "layerdefs": [
        {"layer": 2,
            "n_hid": 80,
            "units": "relu"
        },
        {"layer": 1,
            "input": 9999999999999999
        },
        {"layer": 3,
            "classify": "softmax"
        }
    ],
    "rundefs": {
        "epochs":  24,
        "alpha":   0.74,
        "reg":  "Maxnorm",
        "maxnorm_lim": [4.0],
        "lambda":  0.000191,
        "learn_decay": [0.5,4.0],
        "mb_size_in":   50, 
        "norm_mode":   "none",
        "dobatch": true,
        "do_batch_norm":  true,
        "opt":   "adam",
        "opt_params": [0.9, 0.999],
        "dropout": false,
        "droplim": [1.0,0.8,1.0],
        "plots": ["Train", "Learning", "Test"],
        "plot_now": true,
        "quiet": true,
        "initializer": "xavier",
        "shuffle": false
    }
}
```

**Comment by lewisl** [comment]
Thanks--you are right, of course.  Might replace the first hash with an array.
